pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -147,11 +150,22 @@ func parseConfig(getconfigCtx *getconfigContext, config *zconfig.EdgeDevConfig,
 		parseSystemAdapterConfig(getconfigCtx, config, source, forceSystemAdaptersParse)
 
 		if source != fromBootstrap {
-			parseBaseOS(getconfigCtx, config)
+			activateNewBaseOS := parseBaseOS(getconfigCtx, config)
 			parseNetworkInstanceConfig(getconfigCtx, config)
 			parseContentInfoConfig(getconfigCtx, config)
 			parseVolumeConfig(getconfigCtx, config)
 
+			// We have handled the volumes, so we can now process the app instances. But we need to check if
+			// we are in the middle of a baseOS upgrade, and if so, we need to skip processing the app instances.
+			if (source == fromController && activateNewBaseOS) ||
+				(getconfigCtx.configProcessingSkipFlag == skipConfigUpdate) {
+				// We need to activate the new baseOS
+				// before we can process the app instances
+				// which depend on the new baseOS
+				log.Noticef("parseConfig: Ignoring config as a new baseOS image is being activated")
+				return skipConfigUpdate
```

Comment:
As I understand, once `parseConfig` returns `skipConfigUpdate`, it will start ignoring app config and it will keep doing so until node is rebooted as part of the update procedure, correct?
Then I wonder if there could be a scenario where upgrade procedure fails, e.g. EVE cannot download the new image, and then it stays stuck in the `skipConfigUpdate` state, ignoring app config changes.

Reply:
Yep, the logic is right.
As I understand it, if the EVE update fails, EVE will use the fallback procedure and should not be in the upgrading state anymore. In this case, the config will be not ignored anymore.

Reply:
But if image download for example fails, then there is no need for fallback, is there?
What I don't see is how `parseConfig` will stop returning `skipConfigUpdate` in that case. Maybe I'm just missing something.

Reply:
As far as I remember, even if the EVE image download fails, EVE does not end in the upgrading state. Actually, I can check this with a test.

Reply:
I forced a failure while attempting to download a new EVE image. As a result, the system did not attempt to upgrade and remained online. Overall, I believe that our EVE upgrade process is quite resilient against such problems..

Reply:
Did you see "Ignoring config as a new baseOS image" in the logs?
One more thing I would recommend testing is if after the failure EVE be accepting app config changes (i.e. not returning with `skipConfigUpdate`). For example you can try deploying new app, etc.

Reply:
Yep, I see "Ignoring config as a new baseOS image".
I tested the acceptance of the new config after a successful EVE upgrade.
To test the app upgrade after the EVE upgrade fallback procedure, I have to come up with an idea of how to force the procedure...

Reply:
If the downjoad fails I expect it to wait until the download succeds or it receives a reboot commabd